xilem_core/src/lib.rs

```diff
@@ -50,12 +47,15 @@
 #![expect(clippy::allow_attributes_without_reason, reason = "Deferred: Noisy")]
 extern crate alloc;
 
-// Used only for ad-hoc debugging of tests
-#[cfg(test)]
-extern crate std;
+// // Used only for ad-hoc debugging of tests
+// #[cfg(test)]
+// extern crate std;
```

Comment:
Is this still necessary?

Reply:
The problem with these is that they make Rust-analyzer's auto-complete give you imports from std, even in this `no_std` crate. 
I still don't like that it's not possible to just use e.g. `dbg!` in the tests, but as this does have an actual cost, I'd lean towards removing it entirely.